From 0cb83901803fd12663880dbf4faf29f2628f548e Mon Sep 17 00:00:00 2001
Message-Id: <0cb83901803fd12663880dbf4faf29f2628f548e.1434058173.git.chang-joon.lee@intel.com>
In-Reply-To: <cb13d6abcebb263859514735a399f7f8b895c27a.1434058173.git.chang-joon.lee@intel.com>
References: <cb13d6abcebb263859514735a399f7f8b895c27a.1434058173.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Fri, 22 May 2015 14:15:03 +0530
Subject: [PATCH 5/6] FROM_UPSTREAM [VPG]: drm/i915: Check for panel power only
 for eDP
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

During aux transfers on eDP, we check if the panel is already
powered on; and if not, we do so. But this check should only
be done for eDP and not for DP. Missing this check initiates
Panel power sequence on ports in which DP is connected, resulting
in warnings/blank outs during connected boot scenarios.

This patch fixes it by adding an is_edp() check before checking
whether panel is powered on or not. This patch exists in upstream
through below commit in 3.19 kernel.

Upstream-Commit-id: 1e0560e05db2830f61465ce98b995564d33dfbcc

drm/i915: Rename edp vdd funcs for consistency

edp_* are now the lower level functions and intel_edp_* the higher level
ones. One should use them in pairs.

Reviewed-by: Jani Nikula <jani.nikula@intel.com>
Signed-off-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>

Issue: GMINL-9867
Change-Id: Idffcccb5bdb54968176283fec6b86feef5594bdb
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index ddc74fb..97f03bf 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -634,7 +634,7 @@ intel_dp_aux_ch(struct intel_dp *intel_dp,
 	bool vdd = false;
 
 	/* If we already have panel power, do not call _vdd_on */
-	if (!edp_have_panel_power(intel_dp))
+	if (is_edp(intel_dp) && !edp_have_panel_power(intel_dp))
 		vdd = _edp_panel_vdd_on(intel_dp);
 
 	/* dp aux is extremely sensitive to irq latency, hence request the
-- 
1.9.1

